### PR TITLE
EST with postgresql based authentication

### PR DESCRIPTION
**Pull request by fmarco76**
`PKIPostgreSQLRealm` class has been modified to avoid and change in DB. The DB has to be ready before configuring the realm. A new test working with this realm is included.
    
Additionally, the realm test on separate instance has been modified to  authenticate the EST subsystem into the CA using a certificate.

### CODE REVIEW

**Comment by fmarco76** [comment]
@edewata I'll wait #4854 and update the CI before merge.

**Comment by edewata** [comment]
@fmarco76 I've merged PR #4854. FYI, I'm planning to modify pkispawn so that it can create an SSL server cert for ACME automatically, and I'll modify the ACME test with a separate instance to use that automation, but I'll keep the original test with manual cert creation as a separate test (i.e. ACME with existing NSS database) to make sure it will support reinstallation or containers (where the NSS database already exists).

**Comment by fmarco76** [comment]
@edewata I have updated the actions

File: base/server/src/main/java/com/netscape/cms/realm/PKIPostgreSQLRealm.java
Comment (edewata) [inline base/server/src/main/java/com/netscape/cms/realm/PKIPostgreSQLRealm.java:158]: Is it possible to keep this code? I think it was meant to make ACME easier to use, i.e. just run PostgreSQL and ACME, and it's ready to use (ACME doesn't require initial users). If necessary (e.g. for EST) the tables can be created first and this code will not alter the tables. We probably can add a new param in `realm.conf` to specify the path, for example:
```
create=/usr/share/pki/acme/realm/postgresql/create.sql
```

**Review by edewata — COMMENTED** [review]
It looks good in general, but please see my comments below.

File: base/server/src/main/java/com/netscape/cms/realm/PKIPostgreSQLRealm.java
Comment (fmarco76) [reply base/server/src/main/java/com/netscape/cms/realm/PKIPostgreSQLRealm.java:158]: The problem of this code is that it check for the table and if not present then it will create the tables. If the DB is provided by the user and the table names do not match the new table should not be created. We can make this optional.

File: .github/workflows/est-ds-realm-separate-test.yml
Comment (fmarco76) [reply .github/workflows/est-ds-realm-separate-test.yml:107]: I was looking a way to easily append to  the same p12 but I did not find it. This is way better, thanks!

File: .github/workflows/est-ds-realm-separate-test.yml
Comment (edewata) [inline .github/workflows/est-ds-realm-separate-test.yml:110]: I think we can use `pki ca-user-cert-add --input` so no need to parse the serial number.

https://github.com/dogtagpki/pki/wiki/PKI-User-Certificate-CLI

File: base/server/src/main/java/com/netscape/cms/realm/PKIPostgreSQLRealm.java
Comment (edewata) [inline base/server/src/main/java/com/netscape/cms/realm/PKIPostgreSQLRealm.java:132]: I think the `setup()` was originally called from `connect()` because there is no guarantee that containers will be started in certain orders. For example, sometimes the ACME container is started first, then the PostgreSQL container is added later or it was not running when ACME container is started. With the new code that probably won't work anymore. My suggestion is to return `setup()` to its original position, but we can keep the `dbcreate.file` param. What do you think?

**Review by edewata — APPROVED** [review]
Thanks for the updates! I have a few more comments but I'll let you decide. Feel free to update/merge.

File: base/server/src/main/java/com/netscape/cms/realm/PKIPostgreSQLRealm.java
Comment (fmarco76) [reply base/server/src/main/java/com/netscape/cms/realm/PKIPostgreSQLRealm.java:132]: I do not think this parameter should be used after the installation. If we have it in the init then at the server start the DB should be created and the parameter should be removed. This could be done during pkispawn by the user.

In a container environment there is no reason to provide a DB without table after the initial setup. Who should fill the table with the user? In any case it should be possible to define a startup order although it is not needed in this case.

Since the current implementation perform the check in the `connect()` I am reverting it and evaluate again when we will move to container.

Comment (edewata) [reply base/server/src/main/java/com/netscape/cms/realm/PKIPostgreSQLRealm.java]: Thanks! Yeah, ACME doesn't require any user unless we want to manage something remotely (which there isn't much that can be done anyway). EST seems to require a pre-existing realm, so the requirement is different. We can certainly revisit this requirement again.

**Comment by fmarco76** [comment]
@edewata Thanks!

**Pull request merged**
